### PR TITLE
[FIX] Improper use of exception 

### PR DESCRIPTION
**Pull request by MFizz**
Short Description: 
Exception was used to filter response and additionally catch everything.

### CODE REVIEW

**Comment by cwild** [comment]
Looks messy to me. Can't you just catch ```KeyError``` instead?

**Comment by douglascamata** [comment]
👍 

@cwild much better use `dict.get` than catching `KeyError`, imho.

@TheSavior review please.

File: pokemongo_bot/__init__.py [hunk 1]
```diff
@@ -413,15 +413,11 @@ def current_inventory(self):
         items_stock = {x.value: 0 for x in list(Item)}
 
         for item in inventory_dict:
-            try:
-                # print(item['inventory_item_data']['item'])
-                item_id = item['inventory_item_data']['item']['item_id']
-                item_count = item['inventory_item_data']['item']['count']
-
+            item_id = item.get('inventory_item_data', {}).get('item', {}).get('item_id', {})
+            item_count = item.get('inventory_item_data', {}).get('item', {}).get('count', {})
```
Comment (douglascamata) [inline pokemongo_bot/__init__.py]: better do:

```
item_dict = item.get('inventory_item_data', {}).get('item', {})
item_count = item_dict.get('count')
item_id = item_dict.get('item_id')

if not item_count and not item_id
```

**Comment by MFizz** [comment]
Thx, changed accordingly.

**Comment by TheSavior** [comment]
:+1:

File: pokemongo_bot/__init__.py [hunk 2]
```diff
@@ -413,15 +413,13 @@ def current_inventory(self):
...
         for item in inventory_dict:
-            try:
-                # print(item['inventory_item_data']['item'])
-                item_id = item['inventory_item_data']['item']['item_id']
-                item_count = item['inventory_item_data']['item']['count']
+            item_dict = item.get('inventory_item_data', {}).get('item', {})
+            item_count = item_dict.get('count')
+            item_id = item_dict.get('item_id')
 
+            if not item_count and not item_id:
```
Comment (douglascamata) [inline pokemongo_bot/__init__.py]: I misunderstood the if condition. Previous one was `!= {}`, so the correct is `if item_count and item_id`, my bad.

Comment (MFizz) [inline pokemongo_bot/__init__.py]: I'm sorry, I should have checked too.

**Pull request merged**